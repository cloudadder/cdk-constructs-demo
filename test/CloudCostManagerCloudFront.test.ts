import { App } from 'aws-cdk-lib';
import { Annotations, Template } from 'aws-cdk-lib/assertions';
import { TestStackCloudFront, TestStackWithDatabasePositive } from './TestStacks';

describe('Cloud Cost Manager CloudFront', () => {
  test('(CloudFrontTestPositive): confirm cloud cost management is implemented for database', () => {
    const app = new App();
    var stack = new TestStackCloudFront(app, 'TestStackWithCloudFront');
    Annotations.fromStack(stack).hasInfo('/TestStackWithCloudFront',
      'CloudCostManager validation passed',
    );

    Template.fromStack(stack).hasResourceProperties('AWS::CloudFront::Distribution', {
      DistributionConfig: {
        DefaultCacheBehavior: {
          CachePolicyId: '658327ea-f89d-4fab-a63d-7e88639e58f6',
          Compress: true,
          LambdaFunctionAssociations: [
            {
              EventType: 'viewer-request',
              LambdaFunctionARN: 'arn:aws:lambda:us-east-1:123456789012:function:functionName:1',
            },
          ],
          TargetOriginId: 'TestStackWithCloudFrontTestDistroOrigin1C4B7BCE0',
          ViewerProtocolPolicy: 'allow-all',
        },
        Enabled: true,
        HttpVersion: 'http2',
        IPV6Enabled: true,
        Origins: [
          {
            DomainName: {
              'Fn::GetAtt': [
                'TestBucket560B80BC',
                'RegionalDomainName',
              ],
            },
            Id: 'TestStackWithCloudFrontTestDistroOrigin1C4B7BCE0',
            S3OriginConfig: {
              OriginAccessIdentity: {
                'Fn::Join': [
                  '',
                  [
                    'origin-access-identity/cloudfront/',
                    {
                      Ref: 'TestDistroOrigin1S3Origin20F9517B',
                    },
                  ],
                ],
              },
            },
          },
        ],
        ViewerCertificate: {
          MinimumProtocolVersion: 'TLSv1.2_2021',
        },
      },
      Tags: [
        {
          Key: 'cloud-cost-manager:check:cloudfront',
          Value: 'pass',
        },
        {
          Key: 'cloud-cost-manager:customer-name',
          Value: 'acme-co',
        },
        {
          Key: 'cloud-cost-manager:env-name',
          Value: 'staging',
        },
      ],
    });
  });

  test('Snapshot CloudFront', () => {
    const app = new App();
    const stack = new TestStackWithDatabasePositive(app, 'test');

    expect(app.synth().getStackArtifact(stack.artifactId).template).toMatchSnapshot();
  });
});