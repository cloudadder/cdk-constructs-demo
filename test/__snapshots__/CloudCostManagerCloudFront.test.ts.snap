// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cloud Cost Manager CloudFront Snapshot CloudFront 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "TestBucket560B80BC": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "cloud-cost-manager:customer-name",
            "Value": "acme-co",
          },
          Object {
            "Key": "cloud-cost-manager:env-name",
            "Value": "staging",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "TestBucketPolicyBA12ED38": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "TestBucket560B80BC",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": Object {
                "CanonicalUser": Object {
                  "Fn::GetAtt": Array [
                    "TestDistroOrigin1S3Origin20F9517B",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "TestBucket560B80BC",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "TestDistro62817917": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "DefaultCacheBehavior": Object {
            "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
            "Compress": true,
            "LambdaFunctionAssociations": Array [
              Object {
                "EventType": "viewer-request",
                "LambdaFunctionARN": "arn:aws:lambda:us-east-1:123456789012:function:functionName:1",
              },
            ],
            "TargetOriginId": "testTestDistroOrigin11604225C",
            "ViewerProtocolPolicy": "allow-all",
          },
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "DomainName": Object {
                "Fn::GetAtt": Array [
                  "TestBucket560B80BC",
                  "RegionalDomainName",
                ],
              },
              "Id": "testTestDistroOrigin11604225C",
              "S3OriginConfig": Object {
                "OriginAccessIdentity": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "origin-access-identity/cloudfront/",
                      Object {
                        "Ref": "TestDistroOrigin1S3Origin20F9517B",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "ViewerCertificate": Object {
            "MinimumProtocolVersion": "TLSv1.2_2021",
          },
        },
        "Tags": Array [
          Object {
            "Key": "cloud-cost-manager:check:cloudfront",
            "Value": "pass",
          },
          Object {
            "Key": "cloud-cost-manager:customer-name",
            "Value": "acme-co",
          },
          Object {
            "Key": "cloud-cost-manager:env-name",
            "Value": "staging",
          },
        ],
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "TestDistroOrigin1S3Origin20F9517B": Object {
      "Properties": Object {
        "CloudFrontOriginAccessIdentityConfig": Object {
          "Comment": "Identity for testTestDistroOrigin11604225C",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
